sdk/core/core/inc/az_span.h

```diff
@@ -104,9 +140,120 @@ AZ_NODISCARD AZ_INLINE az_span az_span_from_single_item(uint8_t const * ptr) {
 }
 
 AZ_NODISCARD AZ_INLINE az_span az_str_to_span(char const * str) {
-  return (az_span){ .begin = (uint8_t const *)str, .size = strlen(str) };
+  return (az_span){
+    ._internal = { .begin = (uint8_t const *)str, .capacity = strlen(str), .length = strlen(str) }
+  };
+}
+
+/****************** Mutating az_span (used to be az_mut_span in the origins)     ******/
+
+/**
+ * @brief move the content of span @b src to @buffer and make @b out_result point to it
+ *
+ * @param buffer
+ * @param src
+ * @param out_result
+ * @return AZ_NODISCARD az_mut_span_move
+ */
+AZ_NODISCARD az_result
+az_span_move(az_span const buffer, az_span const src, az_span * const out_result);
```

Comment:
Now that span does carry both capacity and length, all these functions with signatures like `f(az_span const buffer, az_span const src, az_span * const result)` shoud start to look like `f(az_span const src, az_span * const dst)`., so the `dst` is now serves the role of both `buffer` (capacity is not going to change), and `out_result` (dst.length is going to get updated to src.length)